bitcoin/src/address.rs

```diff
@@ -269,17 +271,36 @@ impl<'a> fmt::Display for AddressEncoding<'a> {
             }
             Payload::WitnessProgram(witness_prog) => {
                 let (version, prog) = (witness_prog.version(), witness_prog.program());
-                let mut upper_writer;
-                let writer = if fmt.alternate() {
-                    upper_writer = UpperWriter(fmt);
-                    &mut upper_writer as &mut dyn fmt::Write
-                } else {
-                    fmt as &mut dyn fmt::Write
-                };
-                let mut bech32_writer =
-                    bech32::Bech32Writer::new(self.bech32_hrp, version.bech32_variant(), writer)?;
-                bech32::WriteBase32::write_u5(&mut bech32_writer, version.into())?;
-                bech32::ToBase32::write_base32(&prog.as_bytes(), &mut bech32_writer)
+                match version {
+                    WitnessVersion::V0 => {
+                        for c in prog
+                            .as_bytes()
+                            .iter()
+                            .copied()
+                            .bytes_to_fes()
+                            .with_checksum::<Bech32>(&self.hrp)
+                            .with_witness_version(Fe32::Q)
+                            .chars()
+                        {
+                            fmt.write_char(c)?;
+                        }
+                    }
```

Comment:
In 4321612b0015f9a09ca3f95f2256cdb7ae2d5876:

Lol, we should encapsulate this chain of iterators into a `encode_bytes_fmt::<Ck>` method or something.

Reply:
Yeah I figured that would happen when we touch `bech32/src/lib.rs` or if/when we do a segwit api.

Reply:
Now done, see linked branch in PR description.